File: MobileClient/Class/Shop.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MobileClient.Class
{
    public class Shop
    {
        [JsonIgnore]
        internal static ObservableCollection<Shop> Shops = new ObservableCollection<Shop>();
        [JsonIgnore]
        internal static List<string> ListShops { get { return GetShopNames(); } }

        [JsonProperty("shopName", NullValueHandling = NullValueHandling.Ignore)]
        public string Name { get; set; } = string.Empty;
        [JsonProperty("id", NullValueHandling = NullValueHandling.Ignore)]
        public int Id { get; set; } = int.MinValue;

        [JsonIgnore]
        public string ShowName { get { return $"{this.Id} - {this.Name}"; } }

        public Shop() 
        {
        }

        public Shop(string name, int id)
        {
            Name = name;
            Id = id;
        }

        public string ToJson() 
        {
            return JsonConvert.SerializeObject(this);
        }

        public static Shop FromJson(string json) 
        {
            return JsonConvert.DeserializeObject<Shop>(json);
        }

        public static ObservableCollection<Shop> FromJsonList(string json) 
        {
            return JsonConvert.DeserializeObject<ObservableCollection<Shop>>(json);
        }

        public static string ListToJson(ObservableCollection<Shop> list)
        {
            return JsonConvert.SerializeObject(list);
        }

        public static List<string> GetShopNames() 
        {
            var list = Shops.Select(shop => shop.Name).ToList();
            list.Insert(0,"Wszystkie");
            return list;
        }
    }
}

File: WorkerService/Shop.fs
namespace WorkerService

open Newtonsoft.Json
open System.Collections.ObjectModel

type Shop(id: int, name: string) as self =
    do  
        self.SetValue(id, name)

    member val Name: string = "" with get, set
    member val Id: int = id with get, set

    member this.SetValue(id: int, name: string) =
        this.Id <- id
        this.Name <- name
    
    member this.ToJson() =
        JsonConvert.SerializeObject(self)

    static member FormJson(json: string) : Shop =
        JsonConvert.DeserializeObject<Shop>(json)

    static member SerializeToJson(list: ObservableCollection<Shop>) =
        JsonConvert.SerializeObject(list)

    static member FormJsonShops(json: string) : ObservableCollection<Shop> =
        JsonConvert.DeserializeObject<ObservableCollection<Shop>>(json)

